sgpt/app.py

```diff
@@ -78,37 +92,47 @@ def main(  # pylint: disable=too-many-arguments
         help="Cache completion results.",
     ),
 ) -> None:
+    role_manager.check_and_setup_default_roles()
+    if save_role:
+        role_manager.save_role(save_role, prompt, echo=True)
+        return
+
     if not prompt and not editor:
         raise MissingParameter(param_hint="PROMPT", param_type="string")
 
-    if shell and code:
-        raise BadArgumentUsage("--shell and --code options cannot be used together.")
-
     if editor:
-        prompt = get_edited_prompt()
+        prompt = get_edited_prompt(prompt)
 
     api_host = config.get("OPENAI_API_HOST")
     api_key = config.get("OPENAI_API_KEY")
     client = OpenAIClient(api_host, api_key)
 
     if chat:
-        full_completion = ChatHandler(client, chat, shell, code).handle(
-            prompt,
+        # TODO: Why doesn't execution continue after .handle returns?
```

Comment:
On my system execution doesn't continue after `Handler.handle()` returns. It looks like it goes into some kind of debug trace, but no error message is output to the terminal. 

In any case this stops the prompting for "Execute shell command" on chats.